packages/vuetify/src/effects/elevation/index.ts

```diff
@@ -0,0 +1,22 @@
+// Setup
+import { computed } from 'vue'
+
+// Types
+export interface ElevationProps {
+  elevation?: number
+}
+
+// Props
+export const elevationProps = (
+  defaults: Partial<ElevationProps> = {}
+) => ({
+  elevation: {
+    type: Number,
+    default: defaults.elevation,
+  },
+})
+
+// Effect
+export function useElevationClasses (props: ElevationProps) {
+  return computed(() => (0 ^ props.elevation!) ? { [`elevation-${props.elevation}`]: true } : {})
```

Comment:
I just realized we're just returning the computed directly here. I think it would be better to enforce that we always return an object from composition functions (which is also what I wrote in code guidelines). So this would be:

```
const elevationClasses = computed(...)
return { elevationClasses }

// usage
const { elevationClasses } = useElevationClasses(props)

const classes = computed(() => ({
	...elevationClasses.value,
})
```

Otherwise when we need to return multiple values from a composition function, someone might do this:

```
return computed(() => ({
  foo: ...,
  bar: ...
})

// But when using it you won't be able to destructure it without calling .value
const { foo, bar } = useFoo(props) // won't work
const { foo, bar } = useFoo(props).value // this means we might loose reactivity of foo and bar unless we're careful of what we're returning. also it doesn't look as nice :)

// and if not destructuring
const baz = useFoo(props)

const classes = computed(() => ({
  ...baz.value.foo, // again, doesn't look as nice
})
```

Reply:
This should be resolved with https://github.com/vuetifyjs/vuetify/pull/10315/commits/71069230b7e3dfaa762d78dd3efab10161e5b0f6